spark/v4.0/spark/src/main/java/org/apache/iceberg/spark/data/ParquetWithSparkSchemaVisitor.java

```diff
@@ -154,11 +155,10 @@ public static <T> T visit(DataType sType, Type type, ParquetWithSparkSchemaVisit
         } finally {
           visitor.fieldNames.pop();
         }
-      } else if (sType instanceof VariantType) {
-        // TODO: Use LogicalTypeAnnotation.variantType().equals(annotation) when VARIANT type is
-        // added to Parquet
-        // Preconditions.checkArgument(
-        //  sType instanceof VariantType, "Invalid variant: %s is not a VariantType", sType);
+      } else if (LogicalTypeAnnotation.variantType(Variant.VARIANT_SPEC_VERSION)
+          .equals(annotation)) {
+        Preconditions.checkArgument(
+            sType instanceof VariantType, "Invalid variant: %s is not a VariantType", sType);
```

Comment:
nit: we typically don't use the class name directly in the error msg. maybe sth like this?

```
Invalid variant: Spark type %s is not a variant type
```